### PR TITLE
drivers: auxdisplay: Add driver for PTC PT6314 VFD controller

### PR DESCRIPTION
**Pull request by xingrz**
PT6314 is a VFD controller IC that used by Futaba M202MD15FA module. Datasheet: https://newhavendisplay.com/content/app_notes/PT6314.pdf

The IC supports SPI, I80 and M68 interfaces. This PR implemented the SPI driver.

This PR also added an DTS overlay for the existing auxdisplay sample.

ESP32-C3 DevkitM and M202MD15FA
2f)

### CODE REVIEW

**Review by thedjnK — CHANGES_REQUESTED** [review]
I'm assuming that the board is not "esp" and is something like "esp32" which means that text wrapping from first to second line is not implemented, can you add this?

**Comment by xingrz** [comment]
@thedjnK Thank you for your suggestion. I wasn't aware that this is the expected behavior for the Auxdisplay API. I will update my code later today.

You've reminded me that I may need to allow users to determine certain behaviors through DTS, such as such as column/row shift and line wrapping.

**Comment by xingrz** [comment]
@thedjnK I'm still a bit confused.

PT6314 is like having a 40x2 DDRAM internally, but only a 20x2 area (viewport) is visible. On the hardware level, it automatically wraps the text to the next line when it writes to the 40th character.

I noticed in the Auxdisplay API there are functions like `auxdisplay_cursor_position_set` and `auxdisplay_display_position_set` that deal with the cursor and the visible area. Based on my understanding, I think the driver shouldn't handle line wrapping by itself. It should be up to the user to decide. In my opinion, semantically, the `auxdisplay_write` operation should target the DDRAM rather than the viewport.

**Comment by thedjnK** [comment]
It has a 40x2 internal buffer because it supports 40x2 displays, because you only have a 20x2 display, the other 20 characters are written but not visible on that display. The point in the auxdisplay API is to give an easy API for using displays, I don't really see the point in shifting the requirement to deal with the display size wrapping to the application. The HD44780 driver supports line wrapping which uses the size of the display to handle it, the norktake handles it automatically too, I admit it is not specified in the API documentation but I will have it added.

**Comment by xingrz** [comment]
Thank you for your explanation. I will update my code later.

**Comment by xingrz** [comment]
Updated. Also implemented `cursor_position_get` since we're tracking the cursor in the driver now.

**Comment by kartben** [comment]
> The HD44780 driver supports line wrapping which uses the size of the display to handle it, the norktake handles it automatically too, I admit it is not specified in the API documentation but I will have it added.

Ya it should be either added to the documentation (and maybe turned into an option that people could decide to enable or not?), or just left to the user to implement on their own. 
FWIW the third currently available driver, JHD1313, does not implement wrapping and I didn't even realize it was not the expected behavior since nothing in the documentation was mentioning that text was expected to auto-wrap (and I definitely didn't perceive it as a "bug")

**Comment by xingrz** [comment]
Hi, any progress here? Do I need to make any further changes?

File: drivers/auxdisplay/auxdisplay_pt6314.c [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+
+	return auxdisplay_pt6314_ddram_address_set(dev);
+}
+
+static int auxdisplay_pt6314_cursor_position_get(const struct device *dev, int16_t *x, int16_t *y)
+{
+	struct auxdisplay_pt6314_data *data = dev->data;
+
+	*x = data->cursor_x;
+	*y = data->cursor_y;
```
Comment (thedjnK) [inline drivers/auxdisplay/auxdisplay_pt6314.c:193]: Needs a cast

**Review by thedjnK — COMMENTED** [review]
Minor nit. then good to accept

**Review by thedjnK — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]
LGTM

**Pull request merged**